protocols/secio/src/error.rs

```diff
@@ -31,6 +32,9 @@ pub enum SecioError {
     /// I/O error.
     IoError(IoError),
 
+    /// Protocol buffer error.
+    Protobuf(ProtobufError),
```

Comment:
When can this happen? When we decode data from the remote?

Reply:
In theory `protobuf::Message::write_to_bytes` could return this. Parsing errors are mapped to `Error::HandshakeParsingFailure`. I prefer this `ProtobufError` over `expect`ing successful `write_to_bytes` calls in case protobuf considers more error cases in future releases.